addons/html_editor/controllers/main.py

```diff
@@ -314,37 +314,69 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         return attachment
 
     @http.route(['/web_editor/get_image_info', '/html_editor/get_image_info'], type='jsonrpc', auth='user', website=True)
-    def get_image_info(self, src=''):
+    def get_image_info(self, src='', href_base=''):
         """This route is used to determine the information of an attachment so that
         it can be used as a base to modify it again (crop/optimization/filters).
+        Params:
+            src (str): the src of the image.
+            href_base (str): the url of the page containing the image.
         """
         self._clean_context()
-        attachment = None
-        if src.startswith('/web/image'):
+
+        def _get_attachment_by_url(src):
+            """Find attachment by url. There can be multiple matches because of
+            default snippet images referencing the same image in /static/, so we
+            limit to 1.
+            """
+            return request.env['ir.attachment'].search([
+                '|', ('url', '=like', src), ('url', '=like', '%s?%%' % src),
+                ('mimetype', 'in', list(SUPPORTED_IMAGE_MIMETYPES.keys())),
+            ], limit=1)
+
+        def _extract_attachment_info(attachment):
+            original = (attachment.original_id or attachment)
+            return {
+                'attachment': attachment.read(['id'])[0],
+                'original': original.read(['id', 'image_src', 'mimetype'])[0],
+            }
+
+        # Try to retrieve the original attachment based on this url.
+        attachment = _get_attachment_by_url(src)
+        if attachment:
+            return _extract_attachment_info(attachment)
+
+        # If no match, then search by url *path*. To do so, the src is first
+        # transformed into an absolute url in order to correctly handle relative
+        # url that does not begin with a '/' and protocol relative url.
+        absolute_url = urljoin(href_base, src)
+        path = urlparse(absolute_url).path
+
+        # Probably most common case: the src uses the standard '/web/image'
+        # route.
+        if path.startswith('/web/image'):
```

Comment:
Should this actually work off of the path alone? For all we know this is an URL to e.g. an other Odoo instance.